### PR TITLE
resource/aws_elasticache_replication_group: Prevent re-creation when encryption enabled

### LINKED ISSUE #18075: Secondary Elasticache Cluster will get recreated after each apply
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
```
Terraform v0.14.6, provider v3.32.0
```
### Affected Resource(s)
```
aws_elasticache_replication_group
```
### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# Global
resource "aws_elasticache_global_replication_group" "global" {
  count                              = var.ELASTICACHE ? 1 : 0
  provider                           = aws.primary
  global_replication_group_id_suffix = "XXXXXXXX"
  primary_replication_group_id       = aws_elasticache_replication_group.primary[0].id
}

# Primary
resource "aws_elasticache_replication_group" "primary" {
  count                         = var.ELASTICACHE ? 1 : 0
  provider                      = aws.primary
  replication_group_id          = "XXXXXXXX"
  replication_group_description = "XXXXXXXX"
  engine                        = "redis"
  engine_version                = "6.0.5"#"6.x"
  node_type                     = "cache.m5.large"
  number_cache_clusters         = var.ENVIRONMENT_TYPE == "nonprod" ? 1 : 3
#   parameter_group_name          = "XXXXXX"
  port                          = 6379
  subnet_group_name             = local.primary_elasticache_rg_subnet_group
  security_group_ids            = [XXXXXX, XXXXXX]
  multi_az_enabled              = var.ENVIRONMENT_TYPE == "nonprod" ? false : true
  automatic_failover_enabled    = var.ENVIRONMENT_TYPE == "nonprod" ? false : true
  apply_immediately             = var.ENVIRONMENT_TYPE == "nonprod" ? true : false
  at_rest_encryption_enabled    = true
  transit_encryption_enabled    = true
  auth_token                    = XXXXXX
  snapshot_retention_limit      = X
  snapshot_window               = "XXXXXX"
  final_snapshot_identifier     = "XXXXXXX"
  maintenance_window            = "XXXXXXXX"
}

#Secondary
resource "aws_elasticache_replication_group" "secondary_rg_1" {
  count                         = var.CREATE_SECONDARY_REGION_1 && var.ELASTICACHE ? 1 : 0
  provider                      = aws.secondary_1
  replication_group_id          = "XXXXXX"
  replication_group_description = "XXXXXX"
  global_replication_group_id   = aws_elasticache_global_replication_group.global[0].global_replication_group_id
  number_cache_clusters         = var.ENVIRONMENT_TYPE == "nonprod" ? 1 : 3
#   parameter_group_name          = "XXXXXX"
  port                          = 6379
  subnet_group_name             = local.secondary_elasticache_rg_1_subnet_group
  security_group_ids            = [XXXXX, XXXXX]
  multi_az_enabled              = var.ENVIRONMENT_TYPE == "nonprod" ? false : true
  apply_immediately             = var.ENVIRONMENT_TYPE == "nonprod" ? true : false
  auth_token                    = XXXXXXX
  snapshot_retention_limit      = X
  snapshot_window               = "XXXXX"
  final_snapshot_identifier     = "XXXXXX"
  maintenance_window            = "XXXXXXX"
}
```
### Terraform Plan
```
# aws_elasticache_replication_group.secondary_rg_1[0] must be replaced
-/+ resource "aws_elasticache_replication_group" "secondary_rg_1" {
      ~ arn                            = "xxxxxxxxxxxx" -> (known after apply)
      ~ at_rest_encryption_enabled     = true -> false # forces replacement
      ~ cluster_enabled                = false -> (known after apply)
      + configuration_endpoint_address = (known after apply)
      ~ engine_version                 = "6.0.5" -> (known after apply)
      ~ id                             = "xxxxxxxxxxx" -> (known after apply)
      ~ member_clusters                = [
          - "xxxxxxxx",
        ] -> (known after apply)
      ~ node_type                      = "cache.m5.large" -> (known after apply)
      ~ parameter_group_name           = "xxxxxxxx" -> (known after apply)
      ~ primary_endpoint_address       = "xxxxxxxxx" -> (known after apply)
      ~ reader_endpoint_address        = "xxxxxxxxx" -> (known after apply)
      ~ security_group_names           = [] -> (known after apply)
      - tags                           = {} -> null
      ~ transit_encryption_enabled     = true -> false # forces replacement
        # (17 unchanged attributes hidden)

      ~ cluster_mode {
          ~ num_node_groups         = 1 -> (known after apply)
          ~ replicas_per_node_group = 0 -> (known after apply)
        }
    }
```
### Expected Behavior

Not force a recreation of the resource if nothing has changed

### Actual Behavior

Forces a recreation of the secondary elasticache cluster on every apply

### Steps to Reproduce

1. `terraform apply`

When i provided the encryption attributes the first time i created this secondary cluster, terraform complained about these  attributes conflicting with the **global_replication_group_id** attribute. But the plan doesn't even show this attribute. It's as if it's treating it as a standalone replication group

### PR DESCRIPTION
**Pull request by gdavison**
Makes `at_rest_encryption_enabled` and `transit_encryption_enabled` computed to prevent re-creation of secondary global replication groups.

Closes #18075

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSElasticacheReplicationGroup_\|TestAccAWSElasticacheGlobalReplicationGroup_'

--- PASS: TestAccAWSElasticacheReplicationGroup_Validation_NoNodeType (67.21s)
--- PASS: TestAccAWSElasticacheReplicationGroup_useCmkKmsKeyId (683.82s)
--- PASS: TestAccAWSElasticacheReplicationGroup_clusteringAndCacheNodesCausesError (7.84s)
--- PASS: TestAccAWSElasticacheReplicationGroup_enableAtRestEncryption (741.84s)
--- PASS: TestAccAWSElasticacheReplicationGroup_tags (1018.14s)
--- PASS: TestAccAWSElasticacheReplicationGroup_enableAuthTokenTransitEncryption (965.43s)
--- PASS: TestAccAWSElasticacheReplicationGroup_vpc (1349.09s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_Failover_AutoFailoverEnabled (1603.31s)
--- PASS: TestAccAWSElasticacheReplicationGroup_FinalSnapshot (1630.63s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_MemberClusterDisappears_RemoveMemberCluster_ScaleDown (1709.90s)
--- PASS: TestAccAWSElasticacheReplicationGroup_Uppercase (982.76s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_MemberClusterDisappears_AddMemberCluster (1728.42s)
--- PASS: TestAccAWSElasticacheReplicationGroup_Validation_GlobalReplicationGroupIdAndNodeType (1822.92s)
--- PASS: TestAccAWSElasticacheGlobalReplicationGroup_basic (1845.97s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_Basic (2063.70s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_MemberClusterDisappears_NoChange (2079.60s)
--- PASS: TestAccAWSElasticacheReplicationGroup_updateParameterGroup (1212.17s)
--- PASS: TestAccAWSElasticacheReplicationGroup_enableSnapshotting (1707.43s)
--- PASS: TestAccAWSElasticacheReplicationGroup_multiAz_NoAutomaticFailover (9.18s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_Failover_AutoFailoverDisabled (2525.00s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_MultiAZEnabled (2586.61s)
--- PASS: TestAccAWSElasticacheReplicationGroup_AutomaticFailover_OneCacheCluster (11.85s)
--- PASS: TestAccAWSElasticacheReplicationGroup_updateDescription (1007.73s)
--- PASS: TestAccAWSElasticacheReplicationGroup_updateMaintenanceWindow (1325.06s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_NonClusteredParameterGroup (998.84s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_Basic (1145.04s)
--- PASS: TestAccAWSElasticacheReplicationGroup_disappears (1556.44s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_MemberClusterDisappears_RemoveMemberCluster_AtTargetSize (3264.40s)
--- PASS: TestAccAWSElasticacheReplicationGroup_basic (930.91s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_UpdateNumNodeGroupsAndReplicasPerNodeGroup_ScaleDown (2669.32s)
--- PASS: TestAccAWSElasticacheReplicationGroup_updateNodeSize (2405.21s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_UpdateReplicasPerNodeGroup (1889.93s)
--- PASS: TestAccAWSElasticacheReplicationGroup_multiAzInVpc (1166.72s)
--- PASS: TestAccAWSElasticacheReplicationGroup_GlobalReplicationGroupId_disappears (3778.02s)
--- PASS: TestAccAWSElasticacheGlobalReplicationGroup_disappears (2062.66s)
--- PASS: TestAccAWSElasticacheReplicationGroup_redisClusterInVpc2 (1612.29s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_UpdateNumNodeGroupsAndReplicasPerNodeGroup_ScaleUp (4142.52s)
--- PASS: TestAccAWSElasticacheReplicationGroup_GlobalReplicationGroupId_Basic (4238.09s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_UpdateNumNodeGroups_ScaleUp (2781.37s)
--- PASS: TestAccAWSElasticacheReplicationGroup_multiAzNotInVpc (2047.19s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_UpdateNumNodeGroups_ScaleDown (2999.61s)
--- PASS: TestAccAWSElasticacheGlobalReplicationGroup_Description (2810.83s)
--- PASS: TestAccAWSElasticacheReplicationGroup_GlobalReplicationGroupId_Full (4240.28s)
--- PASS: TestAccAWSElasticacheGlobalReplicationGroup_MultipleSecondaries (5186.50s)
--- PASS: TestAccAWSElasticacheGlobalReplicationGroup_ReplaceSecondary_DifferentRegion (6122.87s)
```

### CODE REVIEW

**Review by bflad — COMMENTED** [review]
Just waiting on the testing -- two quick terrafmt fixes and needs a changelog entry 😄

**Review by bflad — APPROVED** [review]
Looks good other than the previous review comments 🚀

File: aws/resource_aws_elasticache_replication_group_test.go [hunk 1]
```diff
@@ -2624,6 +2668,53 @@ resource "aws_elasticache_replication_group" "primary" {
...
+resource "aws_elasticache_replication_group" "test" {
+  replication_group_id          = "%[1]s-s"
+  replication_group_description = "secondary"
+  global_replication_group_id   = aws_elasticache_global_replication_group.test.global_replication_group_id
+
+  subnet_group_name = aws_elasticache_subnet_group.test.name
+
+  number_cache_clusters = 1
+
+  port                          = 16379
```
[the remark this answers is not in the document]
Comment (gdavison) [reply aws/resource_aws_elasticache_replication_group_test.go:2686]: 🤦

File: aws/resource_aws_elasticache_replication_group_test.go [hunk 2]
```diff
@@ -2624,6 +2668,53 @@ resource "aws_elasticache_replication_group" "primary" {
...
+
+  subnet_group_name = aws_elasticache_subnet_group.primary.name
+
+  node_type = "cache.m5.large"
+
+  engine                = "redis"
+  engine_version        = "5.0.6"
+  number_cache_clusters = 1
+
+  port                          = 6379
```
[the remark this answers is not in the document]
Comment (gdavison) [reply aws/resource_aws_elasticache_replication_group_test.go:2710]: 🤦

**Pull request merged**